contents/javascript/Javascript-framework-VueJs.md

```diff
@@ -151,18 +153,147 @@ VueJs is a relatively new JavaScript framework as compared to Angular and React.
 2. **Language barriers:**<br/>
 A majority of users of VueJs are the Chinese as VueJs is developed by a Chinese American. He is supportive of the Chinese community and hence a lot of the existing plugins are written in Chinese. There might be some language barriers for an English speaking developer seeking for VueJs resources.
 
-#### Detailed Comparison of VueJs with other JavaScript frameworks can be found from:
+-----
+### Main concepts needed in VueJs
+Here are some concepts that you need to start writing Vue code:
+1. **Mutating of data in the DOM**<br/>
+In Vue, the state of the data can be directly modified. Let's say, there is a message in your app. To change the message, insert this line:
+    ```js
+    // Vue way
+    this.message = 'Hello Space';
+    ```
+
+    In React, such a direct modification of data is not allowed, due to React's need to rerun lifecycle hooks after state is being updated. Data can only be updated using `this.setState` method.
+
+    If you have never heard of lifecycle hooks before, think of them as selected points in the program where Vue/React checks the virtual DOM to update it.
+    Here is more information about lifecycle hooks in [React](https://reactjs.org/docs/state-and-lifecycle.html) or [Vue](https://vuejs.org/v2/guide/instance.html#Instance-Lifecycle-Hooks).
+    ```js
+    // React way
+    this.setState({ message: 'Hello Space' });
+    ```
+
+<br>
+
+2. **2-way binding**: `v-model` is used to bind the DOM input field to its data variable. This effectively allows your DOM variables to be automatically in sync with your data, regardless of which one is being updated.
+In other words, this reduces the need for you to manually update your data.
+    ```js
+    <input type="checkbox", v-model=isChecked">
+        <label for="checked">Select</label>
+    ```
+
+    2-way binding is generally used for input form bindings, where new data is given by users and processed to provide other useful data.
+    It is quite similar to the way Angular is currently doing. However, Angular discourages the use of two-way binding.
+
+<br>
+
+3. **1-way data flow**: Data can only be passed from parent to child, via `props`. Props can be of any data type, including Objects. On the other hand, if the child wants to pass data to its parent, `$emit` events have to be used instead.<br/>
+In the code below, `to-do list` is the parent and `item` is the child.
+The data in `item` is being passed to `todo-list` for rendering.
```

Comment:
I don't think the target reader will be able follow this explanation. Is s/he supposed to know the meaning of parent, child, props, etc.?